source/connection-monitoring-and-pooling/tests/README.rst

```diff
@@ -64,10 +64,15 @@ Valid Unit Test Operations are the following:
 
   - ``ms``: The number of milliseconds to sleep the current thread for
 
-- ``waitFor(target)``: wait for thread ``target`` to finish executing. Propagate any errors to the main thread.
+- ``waitForThread(target)``: wait for thread ``target`` to finish executing. Propagate any errors to the main thread.
 
   - ``target``: The name of the thread to wait for.
 
+- ``waitForEvent(event, count)``: block the current thread until ``event`` has occurred ``count`` times
```

Comment:
Should we specify that drivers should use some sort of timeout here and fail otherwise (to avoid infinite hangs e.g. due to an unfair wait queue being tested in `wait-queue-fairness.yml`)?

Reply:
Are there any drivers that implement their test runners without some sort of timeout? I had taken that as a given.

Reply:
Not sure; I mostly meant that we should add a note here that a timeout is generally useful, not actually adding a field for a timeout or anything.